tests/run.sh

```diff
@@ -168,13 +168,19 @@ if ! python3 -m pip show pytest-html >/dev/null; then
 fi
 
 if [[ -n $SPECIFIC_INTEGRATION_TEST ]]; then
-    SPECIFIC_INTEGRATION_TEST_FLAG="-k"
+    python3 -m pytest \
+        $EXTRA_TEST_ARGS \
+        --verbose \
+        --junitxml=results.xml \
+        $HTML_REPORT \
+        "$@" \
+        -k "$SPECIFIC_INTEGRATION_TEST"
+else
+    python3 -m pytest \
+        $EXTRA_TEST_ARGS \
+        --verbose \
+        --junitxml=results.xml \
+        $HTML_REPORT \
+        "$@"
 fi
 
-python3 -m pytest \
-    $EXTRA_TEST_ARGS \
-    --verbose \
-    --junitxml=results.xml \
-    $HTML_REPORT \
-    "$@" \
-    $SPECIFIC_INTEGRATION_TEST_FLAG "$SPECIFIC_INTEGRATION_TEST"
```

Comment:
I know it's fugly, but we could do:
```
    $SPECIFIC_INTEGRATION_TEST_FLAG ${SPECIFIC_INTEGRATION_TEST_FLAG:+"${SPECIFIC_INTEGRATION_TEST}"}
```
instead of the if thingy

Reply:
Oh interesting! I can't say I'm confident the quoting will work correctly there, but I agree it is much nicer. Trying...

Reply:
That worked, nice one!